tests/integration/event_filters_test.go

```diff
@@ -2469,17 +2469,19 @@ func ExpectAllInOrderSequentially(t *testing.T, cmdEvents []cmdEvents, actual *e
 
 	actEvtsCopy := actual.getCopy()
 
+	actEvtIdx := 0
 	// second stage: check events
-	for cmdIdx, cmd := range cmdEvents {
+	for _, cmd := range cmdEvents {
 		syscallsInSets := []string{}
 		checkSets := len(cmd.sets) > 0
 		if checkSets {
 			syscallsInSets = getAllSyscallsInSets(cmd.sets)
 		}
 
 		// compare the expected events with the actual events in the same order
-		for evtIdx, expEvt := range cmd.expectedEvents {
-			actEvt := actEvtsCopy[cmdIdx*len(cmd.expectedEvents)+evtIdx]
+		for _, expEvt := range cmd.expectedEvents {
+			actEvt := actEvtsCopy[actEvtIdx]
+			actEvtIdx++
```

Comment:
So this will go from 0 to some n, and then from n to n+m, and so on... Is this the intended behavior?

Reply:
Yep, as events must be found in order. The index before `cmdIdx*len(cmd.expectedEvents)+evtIdx` could cause an out of bounds access for special events like the created by https://github.com/aquasecurity/tracee/issues/4255#issue-2471922475.

But after your question I realised that it still can cause it. Gonna double check that.

Reply:
Done.